dotCMS/src/main/java/com/dotcms/security/apps/AppsUtil.java

```diff
@@ -0,0 +1,150 @@
+        try (final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(new ByteArrayInputStream(bytes), StandardCharsets.UTF_8))) {
+
+            int chr;
+            while ((chr = reader.read()) != -1) {
+                integers.add(chr);
+            }
+        }
+        return ArrayUtils.toPrimitive(
+                integers.stream().map(value -> (char) value.intValue()).toArray(Character[]::new));
+    }
+
+    /**
+     * This method takes a char array and converts its contents into a byte array No String middle
+     * man is created. https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     * @param chars input
+     * @return byte array
+     */
+    static byte[] charsToBytesUTF(final char[] chars) throws IOException {
+        final CharSequence sequence = java.nio.CharBuffer.wrap(chars);
+        return ByteStreams
+                .toByteArray(new CharSequenceInputStream(sequence, StandardCharsets.UTF_8));
+    }
+
+
+    /**
+     * Encrypt variant of the function of the same name located in The Encryptor util class.
+     * The main difference is that this ones does not use a string in the middle. It directly takes a char array.
+     * And doesn't use a string internally.
+     * @see Encryptor#encrypt(Key, String)
+     * @param key security Key
+     * @param chars
+     * @return encrypted text as a char array
+     * @throws EncryptorException
+     */
+    static char[] encrypt(final Key key, final char[] chars)
+            throws EncryptorException {
+
+        try {
+            final Cipher cipher = Cipher.getInstance(key.getAlgorithm());
+            cipher.init(Cipher.ENCRYPT_MODE, key);
+            final byte[] decryptedBytes = charsToBytesUTF(chars);
+            final byte[] encryptedBytes = cipher.doFinal(decryptedBytes);
+            final String encryptedString = Base64.encode(encryptedBytes);
+            return encryptedString.toCharArray();
+        } catch (Exception e) {
+            throw new EncryptorException(e);
+        }
+    }
+
+    /**
+     * Decrypt variant of the function of the same name located in The Encryptor util class.
+     * The main difference is that this ones does not use a string in the middle to extract the resulting bytes.
+     * @see Encryptor#decrypt(Key, String)
+     * @param key security Key
+     * @param encryptedString
+     * @return decrypted text as a char array
+     * @throws EncryptorException
+     */
+    static char[] decrypt(final Key key, final String encryptedString)
+            throws EncryptorException {
+
+        try {
+            final Cipher cipher = Cipher.getInstance(key.getAlgorithm());
+            cipher.init(Cipher.DECRYPT_MODE, key);
+            final byte[] encryptedBytes = Base64.decode(encryptedString);
+            final byte[] decryptedBytes = cipher.doFinal(encryptedBytes);
+            return bytesToCharArrayUTF(decryptedBytes);
+
+        } catch (Exception e) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)